cpp/src/Ice/ConnectionFactory.cpp

```diff
@@ -489,18 +477,6 @@ IceInternal::OutgoingConnectionFactory::getConnection(
             throw Ice::CommunicatorDestroyedException(__FILE__, __LINE__);
         }
 
-        //
-        // Reap closed connections
-        //
-        vector<Ice::ConnectionIPtr> cons;
-        _monitor->swapReapedConnections(cons);
-        for (const auto& p : cons)
-        {
-            remove(_connections, p->connector(), p);
-            remove(_connectionsByEndpoint, p->endpoint(), p);
-            remove(_connectionsByEndpoint, p->endpoint()->compress(true), p);
```

Comment:
We no longer remove closed connections, it seems this is only clear when the factory is destroyed.